src/request.rs

```diff
@@ -42,16 +57,49 @@ pub struct PotatoRequest {
     pub method: HttpRequestMethod,
     pub path: String,
     pub headers: HashMap<String, String>,
-    pub body: Option<String>,
+    pub body: Option<Vec<u8>>,
 }
 
 impl PotatoRequest {
-    pub fn new(method: HttpRequestMethod, path: &str) -> PotatoRequest {
+    pub fn new(method: HttpRequestMethod, path: &str, body: Option<Vec<u8>>) -> PotatoRequest {
         PotatoRequest {
             method,
             path: path.to_string(),
             headers: HashMap::new(),
-            body: None,
+            body: body,
+        }
+    }
+
+    pub fn from_raw_req(raw: &[u8]) -> PotatoRequest {
+        let index = raw
+            .windows(2)
+            .enumerate()
+            .find(|(_, w)| matches!(*w, b"\r\n"))
+            .map(|(i, _)| i)
+            .unwrap();
+
+        let head = String::from_utf8_lossy(&raw[..index]);
+
+        let bindex = raw
+            .windows(4)
+            .enumerate()
+            .find(|(_, w)| matches!(*w, b"\r\n\r\n"))
+            .map(|(i, _)| i)
+            .unwrap();
+
+        let headers = String::from_utf8_lossy(&raw[index..bindex]);
+        let mut content_len = 0;
+        for i in headers.lines() {
+            if i.is_empty() {
+                continue;
+            }
+            let (a, b) = i.split_at(i.find(":").unwrap());
+            if a.trim().starts_with("Content-Length") {
+                content_len = b.replace(": ", "").to_string().parse::<usize>().unwrap();
+            }
         }
+        let body = raw[bindex + 4..content_len + bindex + 4].to_vec();
+
+        PotatoRequest::new(HttpRequestMethod::GET, &headers, Some(body))
```

Comment:
Why GET?

Reply:
Change to OPTION(any method)

Reply:
You have `head` variable above which should looks something like `[METHOD] [PATH] HTTP/1.1`. You need to extract `METHOD` and `PATH` and set it in `PotatoRequest`